### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by LuKks**
There are more tests pending!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -278,6 +283,18 @@ function alive (list) {
   return true
 }
 
+function isDelayInvalid (ms) {
+  if (ms < 1 || ms > 2147483647) return true
```
Comment (mafintosh) [inline index.js:287]: setTimeout(0) is legal afaik

File: test/helpers/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module.exports = { isAround, sleep }
+
+function isAround (actual, expected, precision = 5) {
+  const diff = Math.abs(actual - expected)
```
Comment (mafintosh) [inline test/helpers/index.js:4]: 5 is prop too little for the macvm on ci (they are very slow).
Also timers guarantee that it never fires too early, ie actualDelay >= expectedDelay always

File: test/helpers/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  return diff <= precision
+}
+
+function sleep (ms) {
+  if (!this.nil) this.nil = new Int32Array(new SharedArrayBuffer(4))
+  Atomics.wait(this.nil, 0, 0, ms)
```
Comment (mafintosh) [inline test/helpers/index.js:10]: Instead of this.nil make a let nil above

File: index.js (hunk 1 above)
Comment (LuKks) [reply index.js:287]: It is, but Node will set a delay of `1` when the value is zero or negative

File: test/helpers/index.js (hunk 2 above)
Comment (LuKks) [reply test/helpers/index.js:4]: True, Mac CI just failed, there is absolutely no way to test a rough precision?

Comment (mafintosh) [reply test/helpers/index.js:4]: No testing timers suck on ci, due to slowness. maybe make the precison depending on process.platform also

File: test/timeout.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+})
+
+test('clearTimeout twice', async function (t) {
+  t.plan(1)
+
+  const id = timers.setTimeout(() => t.fail('timeout should not be called'), 20)
+  setTimeout(() => {
+    timers.clearTimeout(id)
+    timers.clearTimeout(id)
+  }, 15)
```
Comment (mafintosh) [inline test/timeout.js:84]: 5ms lag is prop pushing it here for ci, to like 30ms or something

File: test/timeout.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+  timers.setTimeout(function () {
+    t.pass()
+    throw error
+  }, 10)
+
+  process.once('uncaughtException', function (err) {
+    t.is(err, error)
+  })
+})
```
Comment (mafintosh) [inline test/timeout.js:119]: can you schedule a later timeout here also and see that it triggers, even after the uncaught has triggered?

File: test/timeout.js (hunk 4 above)
Comment (LuKks) [reply test/timeout.js:84]: True! Although the idea is to trigger it before, so I can move the order of the setTimeouts, so it won't matter?

**Pull request merged**